File: 2017.1/2017-06-01/PackageManager/app/src/main/java/br/ufpe/cin/if1001/packagemanager/MainActivity.java
package br.ufpe.cin.if1001.packagemanager;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends Activity {

    public static final String QUERY_KEY = "PackageManagerQuery";
    public static final String GET_PKGS = "getInstalledPackages";
    public static final String GET_APPS = "getInstalledApplications";
    public static final String GET_BROADCASTS = "queryBroadcastReceivers";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button b1 = (Button) findViewById(R.id.btn1);
        b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),LauncherActivity.class));
            }
        });

        Button b2 = (Button) findViewById(R.id.btn2);
        b2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),PrefActivitiesActivity.class));

            }
        });

        Button b3 = (Button) findViewById(R.id.btn3);
        b3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(),PackageManagerQueryActivity.class);
                i.putExtra(QUERY_KEY,GET_PKGS);
                startActivity(i);
            }
        });

        Button b4 = (Button) findViewById(R.id.btn4);
        b4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(),PackageManagerQueryActivity.class);
                i.putExtra(QUERY_KEY,GET_APPS);
                startActivity(i);
            }
        });

        Button b5 = (Button) findViewById(R.id.btn5);
        b5.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(),PackageManagerQueryActivity.class);
                i.putExtra(QUERY_KEY,GET_BROADCASTS);
                startActivity(i);
            }
        });
    }
}

File: 2019-05-13/ContentProvider/app/src/main/java/br/ufpe/cin/android/contentprovider/MainActivity.kt
package br.ufpe.cin.android.contentprovider

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn1.setOnClickListener {
            startActivity(Intent(getApplicationContext(), LerContatosArrayActivity::class.java))
        }

        btn2.setOnClickListener{
            startActivity(Intent(getApplicationContext(), LerContatosCursorActivity::class.java))
        }

        btn3.setOnClickListener{
            startActivity(Intent(getApplicationContext(), LerContatosLoaderActivity::class.java))
        }

        btn4.setOnClickListener {
            startActivity(Intent(getApplicationContext(), ContentConsumerActivity::class.java))
        }

        btn5.setOnClickListener {
            startActivity(Intent(getApplicationContext(), ContentConsumerSQLiteActivity::class.java))
        }

    }
}
